hawkmoth/__init__.py

```diff
@@ -37,6 +37,25 @@ class _AutoBaseDirective(SphinxDirective):
 
     _docstring_types = None
 
+    def __init__(self, *args):
+        super().__init__(*args)
+
+        # Map deprecated options; to be removed.
+        if self.env.config.cautodoc_root:
+            self.env.config.hawkmoth_root = self.env.config.cautodoc_root
+            fmt = 'cautodoc_root option is deprecated, please use hawkmoth_root.'
+            self.logger.warning(fmt, location=(self.env.docname, self.lineno))
```

Comment:
What does `location=(self.env.docname, self.lineno)` reference and does it make sense for this?

Reply:
So it does reference the where the extension is used, but it sure emits a **lot** of warnings.

Reply:
I confess I copied this pattern here. Maybe the other deprecations were less noisy because they were not ubiquitous. E.g.: we don't use transformations everywhere, but we use the root directory for _every_ _single_ _directive_.

Maybe there's some hook in Sphinx in which we could validate `conf.py` variables only once?

Reply:
Looks like the default parameter of `add_config_value()` can be callable: https://www.sphinx-doc.org/en/master/extdev/appapi.html#sphinx.application.Sphinx.add_config_value

This works, but might also be an incredibly stupid thing to do:

```
def _deprecate(conf, old, new, default=None):
    if conf[old]:
        logger = logging.getLogger(__name__)
        logger.warning(f'{old} is deprecated in favour of {new}')
        conf[new] = conf[old]
        del conf[old]
        return conf[new]

    return default

def setup(app):
    ...
    app.add_config_value('hawkmoth_root',
                         lambda conf: _deprecate(conf, 'cautodoc_root', 'hawkmoth_root', app.confdir),
                         'env', [str])
    ...

```

Reply:
Thanks for the snippet! I'll have a look.

Reply:
Going by the docs, I'd say this is pretty neat and an acceptable way of doing this. I'll take this approach in v2.

Only issue I see is that not all deprecations are simple renames, which would complicate this a bit if it mattered in our case. In the limit we'd need one 'deprecation' function for each variable, which I think is ok. We can cross that bridge when we get there of course.